### ISSUE TITLE
Can we have an example of how to use this?

### ISSUE THREAD

**Issue by isaacullah** [jeweinberg/Pandas-MICE#1]
Hi there, I've been looking for a good MICE routine in Pyhton.  I would like to try yours, but you haven't provided the interface or an example, and I can't really tell how to run it by looking at the code. If possible, could you add an example to the readme? Thanks!

**Comment by excetara2** [comment]
I agree

**Comment by jeweinberg** [comment]
Yep I will add an example to the readme file. I wasn't sure if too many would be interested in me making this better! But I will definitely put more effort into it if there is some interest.

**Comment by isaacullah** [comment]
Awesome! Thank you, this will be very helpful.

**Comment by Bill-M123** [comment]
@jeweinberg :  

FWIW: Thanks much for posting this.  I was looking for a straight forward MICE implementation, and if I am using this correctly, I love it.  Here is an example of what I tried.  I assume this close to your intentions?  Also, I think you are spot on with your ideas for improvements, KNN and RF would be useful.

Note: this example is overly simple, but it made it easy to see what was happening.  FWIW:  I needed to seed the nulls to make it work, as that mimicked my data set.

df=pd.DataFrame(columns=['A','B','C','D'])
df['A']=range(1,8)
df['B']=2*df['A']
df['C']=3*df['A']
df['D']=4*df['A']
print('Some data\n',df,'\n')

df1=df.copy()
df1['A'][0]=np.nan
df1['B'][1]=np.nan
df1['C'][2]=np.nan
df1['D'][3]=np.nan
print('Missing Values addedd\n',df1.round(1),'\n')

MICE=MiceImputer()
MICE.seed_nulls=True
df_filled=MICE.fit_transform(df1)
print('Replace nans using MICE:\n',df_filled.round(1),'\n')

df2=df1.fillna(df1.mean())
print('Replace nans with means:\n',df2.round(1),'\n')

print('Difference fill_means - fill_mice\n',df_filled.subtract(df2).round(1))

**Comment by jeweinberg** [comment]
It looks like sklearn has finally implemented a Mice imputer. They call it
IterativeImputer. I am going to suggest you use this package instead, they
would have worked through the use cases in a more robust way than I did.

https://scikit-learn.org/stable/modules/generated/sklearn.impute.IterativeImputer.html
